haptools/sim_genotype.py

```diff
@@ -113,6 +122,9 @@ def output_vcf(
     for ind, sample in enumerate(vcf.samples):
         sample_dict[sample] = ind
 
+    # initialize hap_used array which should contain list of lists for each reference sample's genome and what segment has been used for each
+    haps_used = [[] for samp in range(len(vcf.samples)*2)]
```

Comment:
does this scale well for large numbers of samples? I recall that python allocates memory for list items via some form of exponential backoff, even within list comprehensions
so it might have to repeatedly make copes whilst constructing this list?